### PR TITLE
[patch] dev experience / gha updates

### PR DESCRIPTION
**Pull request by bredmond5**
# Pull Request Template

## Description

- Removes a bunch of extraneous ghas and files
- Creates a single gha that runs tests, and deploys to staging if you merged to master. Tested by commenting out the if statements
- makes startDev.js actually work
- removes all references to the dev/ directory. Just use the dist/
- makes example.html actually get rendered properly

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit test
- [ ] Integration test

## JS Budget Check

Please mention the size in kb before abd after this PR

| Files            | Before      | After       | 
| -----------      | ----------- | ----------- |
| dist/build.js.   |             |             |
| dist/build.min.js|             |             |

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Mentions: 
List the person or team responsible for reviewing proposed changes.

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

File: .github/workflows/build-push.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Install dependencies
+        run: npm run ci
+
+      - name: Lint tests
+        run: npm run lint
+
+      - name: Run tests
+        run: npm run test-report
+
+      - name: Test Report
```
Comment (bredmond5) [inline .github/workflows/build-push.yml:37]: this strange stuff came from the gatekeep.yml

File: .github/workflows/gate-keep.yml
Comment (bredmond5) [inline .github/workflows/gate-keep.yml:1]: i merged gate-keep and deploy-qa into build-push.yml

File: .github/workflows/test.yml
Comment (bredmond5) [inline .github/workflows/test.yml:1]: this was a test gha, it shouldnt have been checked in

File: .gitignore [hunk 2]
```diff
@@ -37,3 +37,6 @@ dev
 test-results.json
 .idea/
 .direnv/
+dev.config
+dev.html
+example.html
```
Comment (bredmond5) [inline .gitignore:42]: 3 generated files

File: Makefile
Comment (bredmond5) [inline Makefile:1]: remove all the references to dev/ directory

File: beta-release.sh
Comment (bredmond5) [inline beta-release.sh:1]: betas are gone

File: src/6_branch.js [hunk 3]
```diff
@@ -1342,16 +1342,16 @@ Branch.prototype['qrCode'] = wrap(
...
 			function(error, rawBuffer) {
 				function QrCode() { }
-
 				if (!error) {
 					QrCode['rawBuffer'] = rawBuffer;
 					QrCode['base64'] = function() {
 						// First Encode array buffer as UTF-8 String, then Base64 Encode
 						if (this['rawBuffer']) {
-							const binaryString = Array.from(new Uint8Array(this.rawBuffer))
+							const binaryString = Array.from(new Uint8Array(rawBuffer))
```
Comment (bredmond5) [inline src/6_branch.js:1350]: well here was the bug

File: startDev.js
Comment (bredmond5) [inline startDev.js:1]: now this script actually spits out a working dev.html, and tells the user to use it. It also builds the websdk so it should always have the latest code

File: .github/workflows/build-push.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Node ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'npm'
+
+      - name: Set up JDK 11
```
Comment (mack-branch) [inline .github/workflows/build-push.yml:22]: where do we use java?

**Review by mack-branch — APPROVED** [review]
lgtm

File: dist/build.js [hunk 5]
```diff
@@ -3111,7 +3111,7 @@ Branch.prototype.qrCode = wrap(callback_params.CALLBACK_ERR_DATA, function(a, b,
     }
     e || (g.rawBuffer = f, g.base64 = function() {
       if (this.rawBuffer) {
-        const h = Array.from(new Uint8Array(this.rawBuffer)).map(k => String.fromCharCode(k)).join("");
+        const h = Array.from(new Uint8Array(f)).map(k => String.fromCharCode(k)).join("");
```
[the remark this answers is not in the document]
Comment (mack-branch) [reply dist/build.js:3114]: nvm see ur other comment

File: .github/workflows/build-push.yml (hunk 4 above)
Comment (bredmond5) [reply .github/workflows/build-push.yml:22]: im not sure why we are setting up java, i was worried about removing it

Comment (bredmond5) [reply .github/workflows/build-push.yml:22]: ill leave for now, but take an action item to look into it

**Pull request merged**